src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTest.cs

```diff
@@ -31,20 +32,26 @@ public class EndpointRoutingIntegrationTest
         public async Task AuthorizationMiddleware_WhenNoAuthMetadataIsConfigured()
         {
             // Arrange
-            var builder = new WebHostBuilder();
-            builder.Configure(app =>
-            {
-                app.UseRouting();
-                app.UseAuthorization();
-                app.UseEndpoints(b => b.Map("/", TestDelegate));
-            })
-           .ConfigureServices(services =>
-           {
-               services.AddAuthorization();
-               services.AddRouting();
-           });
-
-            using var server = new TestServer(builder);
+            var host = new HostBuilder()
+                .ConfigureWebHost(webHostBuilder =>
+                {
+                    webHostBuilder
+                        .Configure(app =>
+                        {
+                            app.UseRouting();
+                            app.UseAuthorization();
+                            app.UseEndpoints(b => b.Map("/", TestDelegate));
+                        })
+                        .UseTestServer();
+                })
+                .ConfigureServices(services =>
+                {
+                    services.AddAuthorization();
+                    services.AddRouting();
+                })
+                .Build();
+
+            using var server = host.GetTestServer();
```

Comment:
Looks like you forgot to start the app. `await host.StartAsync();`

Reply:
I wont comment on the rest of the places as they look like they have the same issue